README.md
The official Appizpanel JavaScript library.

# Appizpanel.js

**Appizpanel.js** is a robust JavaScript client package designed to streamline and enhance asset management in web applications. This dynamic client empowers you to efficiently organize and interact with assets without the need for a static admin panel.

## Key Features:

- **Dynamic Asset Management:** Efficiently organize and manage assets in real-time without relying on a static admin panel.

- **Intuitive Interface:** A user-friendly interface for seamless navigation and interaction with your assets.

- **Live Updates:** Experience instant updates and changes, ensuring that your asset management process is always up-to-date.

- **Customization:** Tailor the client to meet your specific asset management needs with flexible customization options.

## How to Use:

1. **Installation:**
   ```bash
   npm install appizpanel
   ```

CHANGELOG.md
# v0.1.0
 * 
 * 
 * 

test/appizpanel.test.ts
//import Appizpanel from 'appiz-panel'
import Appizpanel from '../src/appizpanel'

describe('Appizpanel Tests', () => {
  describe('when apiKey is provided', () => {
    beforeAll(() => {
      Appizpanel.configure({
        apiKey: 'your_api_key',
        apiVersion: '1.0.0',
        endpointUrl:
          process.env.APPIZPANEL_ENDPOINT_URL || 'https://api.appizpanel.io',
        noRetryIfRateLimited: true,
        requestTimeout: 5000,
      })
    })

    it('should create an instance with configured values', () => {
      const appizpanelInstance = new Appizpanel({
        apiKey: 'your_api_key',
      })
    })
  })

  describe('when apiKey is not provided', () => {
    it('should throw an error during instance creation', () => {
      expect(() => {
        const appizpanelInstance = new Appizpanel()
      }).toThrowError('An API key is required to connect to Appizpanel')
    })
  })
})

describe('when apiKey is not provided', () => {
  it('should throw an error during instance creation', () => {
    // API anahtarı belirtilmediği durumda bir hata almalısınız.
    expect(() => {
      const appizpanelInstance = new Appizpanel()
    }).toThrowError('An API key is required to connect to Appizpanel')

    // API anahtarının geçerli olup olmadığını kontrol etme
    expect(Appizpanel.isApiKeyValid()).toBe(false)
  })
})



lib/appizpanel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
Object.defineProperty(exports, "__esModule", { value: true });
var Appizpanel = /** @class */ (function () {
    function Appizpanel(configs) {
        if (configs === void 0) { configs = {}; }
        var defaultConfig = this.default_config();
        var apiVersion = configs.apiVersion || defaultConfig.apiVersion;
        Appizpanel._apiKey = configs.apiKey || defaultConfig.apiKey || '';
        Appizpanel._apiVersion = apiVersion;
        Appizpanel._apiVersionMajor = apiVersion.split('.')[0];
        Appizpanel._endpointUrl = configs.endpointUrl || defaultConfig.endpointUrl || '';
        Appizpanel._noRetryIfRateLimited = configs.noRetryIfRateLimited || defaultConfig.noRetryIfRateLimited || false;
        Appizpanel._requestTimeout = configs.requestTimeout || defaultConfig.requestTimeout || 0;
        if (!Appizpanel._apiKey) {
            throw new Error(this.getMissingApiKeyErrorMessage());
        }
    }
    Appizpanel.prototype.default_config = function () {
        return {
            apiKey: process.env.APPIZPANEL_API_KEY || '',
            apiVersion: '0.1.0',
            endpointUrl: process.env.APPIZPANEL_ENDPOINT_URL || 'https://api.appizpanel.io',
            noRetryIfRateLimited: false,
            requestTimeout: 300 * 1000,
        };
    };
    Appizpanel.configure = function (config) {
        this._apiKey = config.apiKey || '';
        this._apiVersion = config.apiVersion || '';
        this._apiVersionMajor = config.apiVersion ? config.apiVersion.split('.')[0] : '';
        this._endpointUrl = config.endpointUrl || '';
        this._noRetryIfRateLimited = config.noRetryIfRateLimited || false;
        this._requestTimeout = config.requestTimeout || 0;
    };
    Appizpanel.isApiKeyValid = function () {
        return Appizpanel._apiKey !== '';
    };
    Appizpanel.prototype.getMissingApiKeyErrorMessage = function () {
        return 'An API key is required to connect to Appizpanel';
    };
    return Appizpanel;
}());

exports.default = Appizpanel;


src/appizpanel.ts
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });

export default class Appizpanel {
  private static _apiKey: string;
  private static _apiVersion: string;
  private static _apiVersionMajor: string;
  private static _endpointUrl: string;
  private static _noRetryIfRateLimited: boolean;
  private static _requestTimeout: number;

  constructor(configs: AppizpanelConfig = {}) {
    const defaultConfig = this.default_config();
    const apiVersion: string = configs.apiVersion || defaultConfig.apiVersion!;

    Appizpanel._apiKey = configs.apiKey || defaultConfig.apiKey || '';
    Appizpanel._apiVersion = apiVersion;
    Appizpanel._apiVersionMajor = apiVersion.split('.')[0];
    Appizpanel._endpointUrl = configs.endpointUrl || defaultConfig.endpointUrl || '';
    Appizpanel._noRetryIfRateLimited = configs.noRetryIfRateLimited || defaultConfig.noRetryIfRateLimited || false;
    Appizpanel._requestTimeout = configs.requestTimeout || defaultConfig.requestTimeout || 0;

    if (!Appizpanel._apiKey) {
      throw new Error(this.getMissingApiKeyErrorMessage());
    }
  }

  private default_config(): AppizpanelConfig {
    return {
      apiKey: process.env.APPIZPANEL_API_KEY || '',
      apiVersion: '0.1.0',
      endpointUrl: process.env.APPIZPANEL_ENDPOINT_URL || 'https://api.appizpanel.io',
      noRetryIfRateLimited: false,
      requestTimeout: 300 * 1000,
    };
  }

  public static configure(config: AppizpanelConfig): void {
  
    this._apiKey = config.apiKey || '';
    this._apiVersion = config.apiVersion || '';
    this._apiVersionMajor = config.apiVersion ? config.apiVersion.split('.')[0] : '';
    this._endpointUrl = config.endpointUrl || '';
    this._noRetryIfRateLimited = config.noRetryIfRateLimited || false;
    this._requestTimeout = config.requestTimeout || 0;
  }

  public static isApiKeyValid(): boolean {
    return Appizpanel._apiKey !== '';
  }

  private  getMissingApiKeyErrorMessage(): string {
    return 'An API key is required to connect to Appizpanel';
  }
}

interface AppizpanelConfig {
  apiKey?: string;
  apiVersion?: string;
  endpointUrl?: string;
  noRetryIfRateLimited?: boolean;
  requestTimeout?: number;
}
